packages/integration-karma/test/mixed-shadow-mode/composition/index.spec.js

```diff
@@ -0,0 +1,77 @@
+import { createElement, setFeatureFlagForTest } from 'lwc';
+import Native from 'x/native';
+import NativeRenderFunc from 'x/nativeRenderFunc';
+import SyntheticContainer from 'x/syntheticContainer';
+import NativeContainer from 'x/nativeContainer';
+import NativeLightSynthetic from 'x/nativeLightSynthetic';
+
+if (!process.env.NATIVE_SHADOW && !process.env.COMPAT) {
+    describe('composition', () => {
+        describe('disallow composing synthetic within native', () => {
+            it('basic case', () => {
+                expect(() => {
+                    const elm = createElement('x-native', { is: Native });
+                    document.body.appendChild(elm);
+                }).toThrowError(
+                    'Assert Violation: <x-synthetic> (preferNativeShadow=false) cannot be composed inside of <x-native> (preferNativeShadow=true)'
+                );
+            });
+
+            it('dynamic render', () => {
+                const elm = createElement('x-native-render-func', { is: NativeRenderFunc });
+                document.body.appendChild(elm);
+                expect(elm.shadowRoot.querySelector('div').textContent).toEqual('hello');
+                // The code we want to test calls `Promise.resolve().then(flushCallbackQueue)`, which would result
+                // in an unhandled promise rejection. To actually capture that promise, we mock `Promise.resolve`.
+                let promise = Promise.resolve();
+                spyOn(Promise, 'resolve').and.returnValue({
+                    then: function (callback) {
+                        promise = promise.then(callback);
+                        return promise;
+                    },
+                });
```

Comment:
The canonical way to handle this in LWC is via the [`errorCallback`](https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.create_lifecycle_hooks_error).

Reply:
Done